packages/opencensus-web-instrumentation-zone/test/test-perf-resource-timing-selector.ts

```diff
@@ -0,0 +1,216 @@
+        16.100000011036173,
+        18.100000011036173
+      );
+      // Should ignore this even if the end times are close.
+      const entry4 = createFakePerfResourceEntry(
+        20.100000011036173,
+        30.000000000000003
+      );
+      // Ignore this as its out of the span end time.
+      const entry5 = createFakePerfResourceEntry(
+        35.100000011036173,
+        40.100000011036173
+      );
+      const perfResourceEntries = [entry1, entry2, entry3, entry4, entry5];
+      spyPerfEntryByType(perfResourceEntries);
+      const span = createSpan(12, 30);
+
+      const entriesToBeFiltered = [entry2, entry3];
+      const filteredPerfEntries = getPerfResourceEntries('/test', span);
+      expect(filteredPerfEntries).toEqual(entriesToBeFiltered);
+    });
+  });
+
+  describe('getPossiblePerfResourceEntries', () => {
+    it('Should pair no overlapping entries', () => {
+      const entry1 = createFakePerfResourceEntry(
+        2.100000011036173,
+        5.500000011036173
+      );
+      const entry2 = createFakePerfResourceEntry(
+        5.500000011036173,
+        10.200000011036173
+      );
+      const entry3 = createFakePerfResourceEntry(
+        8.200000011036173,
+        15.500000011036173
+      );
+      const entry4 = createFakePerfResourceEntry(
+        13.500000011036173,
+        20.000000000000003
+      );
+      const perfResourceEntries = [entry1, entry2, entry3, entry4];
+      const filteredPerfEntries = getPossiblePerfResourceEntries(
+        perfResourceEntries
+      );
+
+      expect(filteredPerfEntries.length).toBe(3);
+      // Entry1 and entry2 overlap as entry1 startTime is equal to entry2
+      // endTime.
+      expect(filteredPerfEntries).not.toContain([entry1, entry2]);
+      expect(filteredPerfEntries).toContain([entry1, entry3]);
+      expect(filteredPerfEntries).toContain([entry1, entry4]);
+      expect(filteredPerfEntries).not.toContain([entry2, entry3]);
+      expect(filteredPerfEntries).toContain([entry2, entry4]);
+      expect(filteredPerfEntries).not.toContain([entry3, entry4]);
+    });
+
+    it('Should take single resource timing entries', () => {
+      const entry1 = createFakePerfResourceEntry(
+        2.100000011036173,
+        5.500000011036173
+      );
+      const entry2 = createFakePerfResourceEntry(
+        5.500000011036173,
+        10.200000011036173
+      );
+      const entry3 = createFakePerfResourceEntry(
+        1.500000011036173,
+        20.000000000000003
+      );
+      const perfResourceEntries = [entry1, entry2, entry3];
+      const filteredPerfEntries = getPossiblePerfResourceEntries(
+        perfResourceEntries
+      );
+      expect(filteredPerfEntries.length).toBe(3);
+      expect(filteredPerfEntries).toContain(entry1);
+      expect(filteredPerfEntries).toContain(entry2);
+      expect(filteredPerfEntries).toContain(entry3);
+    });
+
+    it('Should take single and tuples of resource timing entries', () => {
```

Comment:
How would this test change if we make both cases use arrays? Is there still a useful test case here?